src/submission/validation-clinical/donor.ts

```diff
@@ -75,22 +96,171 @@ function checkTimeConflictWithSpecimens(
-  if (specimenIdsWithTimeConflicts.length > 0) {
-    errors.push(
-      utils.buildSubmissionError(
-        donorRecord,
-        DataValidationErrors.CONFLICTING_TIME_INTERVAL,
-        DonorFieldsEnum.survival_time,
-        {
-          conflictingSpecimenSubmitterIds: specimenIdsWithTimeConflicts,
-        },
-      ),
+  return specimenIdsWithTimeConflicts.length > 0
+    ? [
+        utils.buildSubmissionError(
+          donorRecord,
+          DataValidationErrors.CONFLICTING_TIME_INTERVAL,
+          DonorFieldsEnum.survival_time,
+          {
+            conflictingSpecimenSubmitterIds: specimenIdsWithTimeConflicts,
+          },
+        ),
+      ]
+    : [];
+}
+
+const crossFileValidator = async (
+  submittedDonorRecord: DeepReadonly<SubmittedClinicalRecord>,
+  submittedRecords: DeepReadonly<ClinicalSubmissionRecordsByDonorIdMap>,
+  mergedDonor: DeepReadonly<Donor>,
+  // submitted records needed to validate current submission
+) => {
+  const { lost_to_followup_after_clinical_event_id, program_id } = submittedDonorRecord;
+  const { primaryDiagnoses, treatments, followUps } = mergedDonor;
+  const programShortName = program_id as string;
+  const errors: SubmissionValidationError[] = [];
+
+  // Compare across other Submissions
+  let submittedTreatments: DeepReadonly<ClinicalInfo>[] = [];
+  let submittedPrimaryDiagnoses: DeepReadonly<ClinicalInfo>[] = [];
+  let submittedFollowUps: DeepReadonly<ClinicalInfo>[] = [];
+
+  for (const donorId in submittedRecords) {
+    const submittedTreatmentRecords = submittedRecords[donorId].treatment;
+    const submittedPrimaryDiagnosisRecords = submittedRecords[donorId].primaryDiagnoses;
+    const submittedFollowUpRecords = submittedRecords[donorId].followUps;
+    if (submittedTreatmentRecords && submittedTreatmentRecords.length)
+      submittedTreatments = [...submittedTreatments, ...submittedTreatmentRecords];
+
+    if (submittedPrimaryDiagnosisRecords && submittedPrimaryDiagnosisRecords.length)
+      submittedPrimaryDiagnoses = [
+        ...submittedPrimaryDiagnoses,
+        ...submittedPrimaryDiagnosisRecords,
+      ];
+
+    if (submittedFollowUpRecords && submittedFollowUpRecords.length)
+      submittedFollowUps = [...submittedFollowUps, ...submittedFollowUpRecords];
+  }
+
+  // Compare across all Donors in DB
+  const programId = 'TEST-CA';
+  const query: ClinicalQuery = {
+    programShortName,
+    entityTypes: [
+      ClinicalEntitySchemaNames.PRIMARY_DIAGNOSIS,
+      ClinicalEntitySchemaNames.TREATMENT,
+      ClinicalEntitySchemaNames.FOLLOW_UP,
+    ],
+    page: 0,
+    sort: 'donorId',
+    donorIds: [],
+    submitterDonorIds: [],
+  };
+
+  const { donors } = await donorDao.findByPaginatedProgramId(programId, query);
+
+  const storedTreatments = donors.map(donor => donor.treatments).filter(donor => Boolean(donor));
+  const storedPrimaryDiagnoses = donors
+    .map(donor => donor.primaryDiagnoses)
+    .filter(donor => Boolean(donor));
```

Comment:
I'm not sure about the intent of this logic. 

The `filter` is getting an array of `primaryDiagnoses` not the `donor`.
Unclear on what the filter is doing as it's likely always true? `Boolean({})` and `Boolean(string[])` will be `true`.